modules/custom/dkan_common/src/Service/Factory.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Drupal\dkan_common\Service;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Sae\Sae;
+use Contracts\Storage as ContractsStorageInterface;
+
+/**
+ * Factory to generate DKAN API responses.
+ */
+class Factory implements ContainerInjectionInterface {
```

Comment:
Are either of these services used anywhere but in the dkan_api module? Why do you think is worth separating them?

Reply:
It's to wrap the `new` classes being instantiated. If you use `new` to create classes in the regular code, you can't write tests for them.

The interra_api module seems to reference the `JsonREsponse` class. `Sae` is used in `dkan_api` and `dkan_harvest`